### PR TITLE
Introduce a exclude filter for params and/or response

### PR DESCRIPTION
**Pull request by mnin**
We need to prevent the transformation for one specific request in our project.

This PR give us the possibility to exclude some requests by match something from the `env` variable in a `proc` like this:

```ruby
  config.middleware.use OliveBranch::Middleware, exclude_params: -> (env) { ENV['PATH_INFO'].match(/^\/do_not_transform/) }
```

### CODE REVIEW

File: lib/olive_branch/middleware.rb [hunk 1]
```diff
@@ -71,6 +78,14 @@ def call(env)
 
     private
 
+    def exclude_params?(env)
+      !@content_type_check.call(env["CONTENT_TYPE"]) || @exclude_params.call(env)
+    end
+
+    def exclude_response?(env, headers)
+      !@content_type_check.call(headers["Content-Type"]) || @exclude_response.call(env)
```
Comment (efatsi) [inline lib/olive_branch/middleware.rb]: organizational preference here would be to break the content type checks into separate methods. conceptually i think there's a disconnect between excluding params/response and checking for applicable content types. would probably go with something like this up top:
```ruby
if inflection && valid_content_type(env["CONTENT_TYPE"]) && !exclude_params(env)
``
makes the conditional longer but i think reads better

**Comment by efatsi** [comment]
Thanks for the PR, it looks great. Left one little comment about method organization, let me know if you agree/disagree, happy to get this merged in shortly.

**Comment by mnin** [comment]
@efatsi I moved the `inflection` check into the `exclude_` methods, I think it is better readable now

File: README.md [hunk 2]
```diff
@@ -46,18 +46,38 @@ end
...
+#### Excluding URLs
+
+Additionally you can define a custom check by passing a proc
+
+For params transforming
 
 ```ruby
-    config.middleware.use OliveBranch::Middleware, inflection: 'camel'
+    config.middleware.use OliveBranch::Middleware, exclude_params: -> (env) { ENV['PATH_INFO'].match(/^\/do_not_transform/) }
+```
```
Comment (KingHenne) [inline README.md]: typo: `ENV` should be `env`

File: README.md [hunk 3]
```diff
@@ -46,18 +46,38 @@ end
...
+
+Or response transforming
+
+```ruby
+    config.middleware.use OliveBranch::Middleware, exclude_response: -> (env) { ENV['PATH_INFO'].match(/^\/do_not_transform/) }
```
Comment (KingHenne) [inline README.md]: typo: `ENV` should be `env`

**Comment by mnin** [comment]
@KingHenne thanks for reviewing it!

**Comment by efatsi** [comment]
Great stuff.

**Comment by efatsi** [comment]
Deployed version 2.1.2. Thanks again for the work here folks.

**Pull request merged**